// @buffetjs/core/dist/components/ListHeader/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _Button = _interopRequireDefault(require("../Button"));

var _Flex = _interopRequireDefault(require("../Flex"));

var _Padded = _interopRequireDefault(require("../Padded"));

var _Text = _interopRequireDefault(require("../Text"));

var _BaselineAlignement = _interopRequireDefault(require("./BaselineAlignement"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ListHeader(_ref) {
  var button = _ref.button,
      title = _ref.title,
      subtitle = _ref.subtitle;
  return _react["default"].createElement(_styles.ListHeader, null, _react["default"].createElement(_Flex["default"], {
    justifyContent: "space-between"
  }, _react["default"].createElement(_Padded["default"], {
    top: true,
    bottom: true,
    size: "smd"
  }, _react["default"].createElement(_Text["default"], {
    fontSize: "lg",
    fontWeight: "bold",
    lineHeight: "18px"
  }, title), subtitle && _react["default"].createElement(_Padded["default"], {
    top: true,
    size: "xs"
  }, _react["default"].createElement(_Text["default"], {
    fontSize: "md",
    color: "grey",
    lineHeight: "13px"
  }, subtitle))), button && _react["default"].createElement(_BaselineAlignement["default"], null, _react["default"].createElement(_Button["default"], button))));
}

ListHeader.defaultProps = {
  button: null,
  title: null,
  subtitle: null
};
ListHeader.propTypes = {
  button: _propTypes["default"].shape({
    color: _propTypes["default"].string,
    icon: _propTypes["default"].bool,
    type: _propTypes["default"].string
  }),
  subtitle: _propTypes["default"].string,
  title: _propTypes["default"].string
};
var _default = ListHeader;
exports["default"] = _default;
