// @buffetjs/core/dist/components/Checkbox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _Label = _interopRequireDefault(require("../Label"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Checkbox(_ref) {
  var autoFocus = _ref.autoFocus,
      className = _ref.className,
      id = _ref.id,
      message = _ref.message,
      name = _ref.name,
      onChange = _ref.onChange,
      tabIndex = _ref.tabIndex,
      value = _ref.value,
      rest = _objectWithoutProperties(_ref, ["autoFocus", "className", "id", "message", "name", "onChange", "tabIndex", "value"]);

  var handleChange = function handleChange() {
    var target = {
      name: name,
      type: 'checkbox',
      value: !value
    };
    onChange({
      target: target
    });
  };

  return _react["default"].createElement(_styles.CheckboxWrapper, {
    className: className,
    disabled: rest.disabled
  }, _react["default"].createElement(_styles.Checkbox, _extends({}, rest, {
    autoFocus: autoFocus,
    id: id || name,
    name: name,
    onChange: handleChange,
    tabIndex: tabIndex,
    checked: value
  })), _react["default"].createElement(_Label["default"], {
    htmlFor: id || name,
    message: message
  }));
}

Checkbox.defaultProps = {
  autoComplete: 'off',
  autoFocus: false,
  className: null,
  disabled: false,
  id: null,
  message: null,
  onChange: function onChange() {},
  placeholder: null,
  tabIndex: '0',
  value: false
};
Checkbox.propTypes = {
  autoComplete: _propTypes["default"].string,
  autoFocus: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  disabled: _propTypes["default"].bool,
  id: _propTypes["default"].string,
  message: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].string, _propTypes["default"].shape({
    id: _propTypes["default"].string,
    params: _propTypes["default"].object
  })]),
  name: _propTypes["default"].string.isRequired,
  onChange: _propTypes["default"].func,
  placeholder: _propTypes["default"].string,
  tabIndex: _propTypes["default"].string,
  value: _propTypes["default"].bool
};
var _default = Checkbox;
exports["default"] = _default;
