// @buffetjs/core/build/esm/components/HeaderTitle/index.js
/**
 *
 * HeaderTitle
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import { upperFirst } from 'lodash';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons';
import { HeaderTitle as Wrapper } from '@buffetjs/styles';

function HeaderTitle(_ref) {
  var title = _ref.title,
      children = _ref.children,
      cta = _ref.cta;
  var content = upperFirst(title) || children;

  var renderCTA = function renderCTA() {
    if (cta) {
      var onClick = cta.onClick;
      return /*#__PURE__*/React.createElement(FontAwesomeIcon, {
        icon: faPencilAlt,
        onClick: onClick,
        role: "button",
        tabIndex: "0",
        size: "sm"
      });
    }

    return null;
  };

  return /*#__PURE__*/React.createElement(Wrapper, null, /*#__PURE__*/React.createElement("h1", null, content, renderCTA()));
}

HeaderTitle.defaultProps = {
  children: null,
  cta: null,
  title: null
};
HeaderTitle.propTypes = {
  children: PropTypes.node,
  cta: PropTypes.shape({
    onClick: PropTypes.func
  }),
  title: PropTypes.string
};
export default HeaderTitle;
