// @buffetjs/core/dist/components/HeaderTitle/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodash = require("lodash");

var _reactFontawesome = require("@fortawesome/react-fontawesome");

var _freeSolidSvgIcons = require("@fortawesome/free-solid-svg-icons");

var _styles = require("@buffetjs/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function HeaderTitle(_ref) {
  var title = _ref.title,
      children = _ref.children,
      cta = _ref.cta;
  var content = (0, _lodash.upperFirst)(title) || children;

  var renderCTA = function renderCTA() {
    if (cta) {
      var onClick = cta.onClick;
      return _react["default"].createElement(_reactFontawesome.FontAwesomeIcon, {
        icon: _freeSolidSvgIcons.faPencilAlt,
        onClick: onClick,
        role: "button",
        tabIndex: "0",
        size: "sm"
      });
    }

    return null;
  };

  return _react["default"].createElement(_styles.HeaderTitle, null, _react["default"].createElement("h1", null, content, renderCTA()));
}

HeaderTitle.defaultProps = {
  children: null,
  cta: null,
  title: null
};
HeaderTitle.propTypes = {
  children: _propTypes["default"].node,
  cta: _propTypes["default"].shape({
    onClick: _propTypes["default"].func
  }),
  title: _propTypes["default"].string
};
var _default = HeaderTitle;
exports["default"] = _default;
