// @buffetjs/core/dist/components/InputNumber/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _rcInputNumber = _interopRequireDefault(require("rc-input-number"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function InputNumber(_ref) {
  var autoFocus = _ref.autoFocus,
      id = _ref.id,
      className = _ref.className,
      name = _ref.name,
      onChange = _ref.onChange,
      tabIndex = _ref.tabIndex,
      value = _ref.value,
      rest = _objectWithoutProperties(_ref, ["autoFocus", "id", "className", "name", "onChange", "tabIndex", "value"]);

  var handleChange = function handleChange(data) {
    if (data !== null) {
      var target = {
        id: id,
        name: name,
        type: 'number',
        value: data
      };
      onChange({
        target: target
      });
    }
  };

  return _react["default"].createElement(_styles.InputNumber, {
    className: className
  }, _react["default"].createElement(_rcInputNumber["default"], _extends({}, rest, {
    autoFocus: autoFocus,
    id: id || name,
    name: name,
    onChange: handleChange,
    tabIndex: tabIndex,
    value: value,
    className: "inputNumber"
  })));
}

InputNumber.defaultProps = {
  autoComplete: 'off',
  autoFocus: false,
  className: null,
  id: null,
  onChange: function onChange() {},
  placeholder: null,
  tabIndex: '0',
  value: 0
};
InputNumber.propTypes = {
  autoComplete: _propTypes["default"].string,
  autoFocus: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  id: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  onChange: _propTypes["default"].func,
  placeholder: _propTypes["default"].string,
  tabIndex: _propTypes["default"].string,
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number])
};
var _default = InputNumber;
exports["default"] = _default;
