// @buffetjs/core/dist/components/Label/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodash = require("lodash");

var _styles = require("@buffetjs/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Label(props) {
  var content = function () {
    if ((0, _lodash.isFunction)(props.message)) {
      return props.message();
    }

    if (props.message) {
      return props.message;
    }

    return props.children;
  }();

  return _react["default"].createElement(_styles.Label, {
    htmlFor: props.htmlFor,
    className: props.className
  }, content);
}

Label.defaultProps = {
  children: null,
  className: null,
  message: null
};
Label.propTypes = {
  children: _propTypes["default"].node,
  className: _propTypes["default"].string,
  htmlFor: _propTypes["default"].string.isRequired,
  message: _propTypes["default"].oneOfType([_propTypes["default"].func, _propTypes["default"].string, _propTypes["default"].shape({
    id: _propTypes["default"].string,
    params: _propTypes["default"].object
  })])
};
var _default = Label;
exports["default"] = _default;
