// @buffetjs/core/build/esm/components/Picker/PickerSection.js
var _templateObject;

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import styled from 'styled-components';
import PropTypes from 'prop-types';
var PickerSection = styled.div(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  display: none;\n  position: absolute;\n  top: 38px;\n  left: ", ";\n  right: ", ";\n  z-index: 1046;\n  background-color: ", ";\n  border-radius: ", ";\n  border: 1px solid ", ";\n  box-shadow: 0 2px 4px ", ";\n  ", "\n"])), function (_ref) {
  var position = _ref.position;
  return position === 'left' ? '0' : 'auto';
}, function (_ref2) {
  var position = _ref2.position;
  return position === 'right' ? '0' : 'auto';
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.main.colors.white;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.main.sizes.borderRadius;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.main.colors.darkGrey;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.main.colors.greyAlpha;
}, function (_ref7) {
  var isOpen = _ref7.isOpen;
  return isOpen && 'display: block;';
});
PickerSection.defaultProps = {
  isOpen: false,
  position: 'left',
  // TODO : This is temporary
  theme: {
    main: {
      colors: {
        white: '#ffffff',
        darkGrey: '#e3e9f3',
        greyAlpha: 'rgba(227, 233, 243, 0.5)'
      },
      sizes: {
        borderRadius: '2px'
      }
    }
  }
};
PickerSection.propTypes = {
  isOpen: PropTypes.bool,
  position: PropTypes.oneOf(['left', 'right'])
};
export default PickerSection;
