// @buffetjs/core/dist/components/Paging/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactstrap = require("reactstrap");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _Icon = _interopRequireDefault(require("../Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Paging(_ref) {
  var count = _ref.count,
      page = _ref.page,
      onChange = _ref.onChange,
      range = _ref.range;
  var pagesCount = Math.ceil(count / range);
  return _react["default"].createElement(_react["default"].Fragment, null, _react["default"].createElement(_styles.Paging, null), _react["default"].createElement(_reactstrap.Pagination, {
    "aria-label": "Page navigation"
  }, _react["default"].createElement(_reactstrap.PaginationItem, null, _react["default"].createElement(_reactstrap.PaginationLink, {
    onClick: onChange,
    value: 0
  }, _react["default"].createElement(_Icon["default"], {
    icon: "left"
  }))), Array(pagesCount).fill(1).map(function (item, index) {
    return _react["default"].createElement(_react["default"].Fragment, {
      key: index
    }, _react["default"].createElement(_reactstrap.PaginationItem, {
      key: index,
      onClick: onChange,
      className: page === index * range ? 'selected' : ''
    }, _react["default"].createElement(_reactstrap.PaginationLink, {
      onClick: onChange,
      value: index * range
    }, index + 1)));
  }), _react["default"].createElement(_reactstrap.PaginationItem, null, _react["default"].createElement(_reactstrap.PaginationLink, {
    onClick: onChange,
    value: (pagesCount - 1) * range
  }, _react["default"].createElement(_Icon["default"], {
    icon: "right"
  })))));
}

Paging.defaultProps = {
  count: 0,
  onChange: function onChange() {},
  page: 0,
  range: 15
};
Paging.propTypes = {
  count: _propTypes["default"].number,
  onChange: _propTypes["default"].func,
  page: _propTypes["default"].number,
  range: _propTypes["default"].number
};
var _default = Paging;
exports["default"] = _default;
