// @buffetjs/core/dist/components/Padded/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  padding-top: ", ";\n  padding-right: ", ";\n  padding-bottom: ", ";\n  padding-left: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Padded = _styledComponents["default"].div(_templateObject(), function (_ref) {
  var theme = _ref.theme,
      size = _ref.size,
      top = _ref.top;
  return top && (theme.main.sizes.paddings[size] || size);
}, function (_ref2) {
  var theme = _ref2.theme,
      size = _ref2.size,
      right = _ref2.right;
  return right && (theme.main.sizes.paddings[size] || size);
}, function (_ref3) {
  var theme = _ref3.theme,
      size = _ref3.size,
      bottom = _ref3.bottom;
  return bottom && (theme.main.sizes.paddings[size] || size);
}, function (_ref4) {
  var theme = _ref4.theme,
      size = _ref4.size,
      left = _ref4.left;
  return left && (theme.main.sizes.paddings[size] || size);
});

Padded.defaultProps = {
  bottom: false,
  left: false,
  right: false,
  size: 'sm',
  top: false,
  theme: {
    main: {
      sizes: {
        paddings: {
          xs: '5px',
          sm: '10px',
          smd: '20px',
          md: '30px',
          lg: '40px'
        }
      }
    }
  }
};
Padded.propTypes = {
  bottom: _propTypes["default"].bool,
  left: _propTypes["default"].bool,
  right: _propTypes["default"].bool,
  size: _propTypes["default"].string,
  theme: _propTypes["default"].object,
  top: _propTypes["default"].bool
};
var _default = Padded;
exports["default"] = _default;
