// @buffetjs/core/build/esm/components/Count/Wrapper.js
var _templateObject;

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

import styled from 'styled-components';
import PropTypes from 'prop-types';
import Flex from '../Flex';
import colors from '../../theme/colors';
var Wrapper = styled(Flex)(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  width: fit-content;\n  height: 1.4rem;\n  min-width: 1.4rem;\n  padding: 0 0.2rem;\n  border-radius: ", ";\n  background-color: ", ";\n"])), function (_ref) {
  var theme = _ref.theme;
  return theme.main.sizes.borderRadius;
}, function (_ref2) {
  var theme = _ref2.theme,
      isActive = _ref2.isActive;
  return isActive ? theme.main.colors.lightBlue : '#e9eaeb';
});
Wrapper.defaultProps = {
  theme: {
    main: {
      colors: colors,
      sizes: {
        borderRadius: '2px',
        paddings: {
          xs: '5px'
        }
      }
    }
  }
};
Wrapper.propTypes = {
  isActive: PropTypes.bool,
  theme: PropTypes.object
};
export default Wrapper;
