// @buffetjs/core/dist/components/Button/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _icons = require("@buffetjs/icons");

var _Flex = _interopRequireDefault(require("../Flex"));

var _PrefixIcon = _interopRequireDefault(require("../PrefixIcon"));

var _Spinner = _interopRequireDefault(require("./Spinner"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function Button(_ref) {
  var children = _ref.children,
      color = _ref.color,
      disabled = _ref.disabled,
      icon = _ref.icon,
      label = _ref.label,
      isLoading = _ref.isLoading,
      rest = _objectWithoutProperties(_ref, ["children", "color", "disabled", "icon", "label", "isLoading"]);

  var content = label || children;
  var img = icon === true ? _react["default"].createElement(_icons.Plus, {
    height: "11px",
    width: "11px",
    fill: "".concat(_styles.colors.button[color].color)
  }) : _react["default"].createElement(_PrefixIcon["default"], {
    icon: icon
  });

  if (isLoading) {
    return _react["default"].createElement(_styles.Button, _extends({
      color: color,
      disabled: disabled
    }, rest, {
      onClick: function onClick(e) {
        e.preventDefault();
      }
    }), _react["default"].createElement(_Flex["default"], {
      justifyContent: "space-around"
    }, _react["default"].createElement(_Spinner["default"], null)));
  }

  return _react["default"].createElement(_styles.Button, _extends({
    color: color,
    disabled: disabled
  }, rest), img, content);
}

Button.defaultProps = {
  children: null,
  color: 'primary',
  disabled: false,
  icon: false,
  isLoading: false,
  label: null,
  type: 'button'
};
Button.propTypes = {
  children: _propTypes["default"].node,
  color: _propTypes["default"].oneOf(['primary', 'secondary', 'cancel', 'success', 'delete', 'none']),
  disabled: _propTypes["default"].bool,
  icon: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].bool]),
  isLoading: _propTypes["default"].bool,
  label: _propTypes["default"].string,
  type: _propTypes["default"].oneOf(['submit', 'reset', 'button', 'menu'])
};
var _default = Button;
exports["default"] = _default;
