// @buffetjs/core/build/esm/components/NavTabs/index.js
/**
 *
 * NavTabs
 *
 */
import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Tab as StyledTab, colors } from '@buffetjs/styles';
import { gradient } from '@buffetjs/utils';

function NavTabs(_ref) {
  var links = _ref.links;
  var color = colors.greyLink;
  return /*#__PURE__*/React.createElement("ul", null, links.map(function (link, index) {
    color = gradient(color, 1.5);
    var content = link.label ? /*#__PURE__*/React.createElement("p", null, link.label) : link.compo || null;
    return /*#__PURE__*/React.createElement(StyledTab, {
      key: link.to,
      index: index,
      count: links.length,
      background: color
    }, /*#__PURE__*/React.createElement(NavLink, {
      to: link.to
    }, content));
  }));
}

NavTabs.defaultProps = {
  links: []
};
NavTabs.propTypes = {
  links: PropTypes.arrayOf(PropTypes.shape({
    compo: PropTypes.node,
    label: PropTypes.string,
    to: PropTypes.string
  }))
};
export default NavTabs;
