// @buffetjs/core/dist/components/List/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _ListRow = _interopRequireDefault(require("../ListRow"));

var _Padded = _interopRequireDefault(require("../Padded"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function List(_ref) {
  var className = _ref.className,
      items = _ref.items,
      isLoading = _ref.isLoading,
      customRowComponent = _ref.customRowComponent;
  return _react["default"].createElement(_styles.List, {
    className: className
  }, isLoading ? _react["default"].createElement(_Padded["default"], {
    top: true,
    bottom: true,
    size: "md"
  }, _react["default"].createElement(_styles.LoadingIndicator, null)) : _react["default"].createElement("table", null, _react["default"].createElement("tbody", null, items.map(function (item, index) {
    return customRowComponent ? _react["default"].createElement(_react["default"].Fragment, {
      key: index
    }, customRowComponent(item)) : _react["default"].createElement(_ListRow["default"], {
      cells: item,
      key: index
    });
  }))));
}

List.defaultProps = {
  className: null,
  customRowComponent: null,
  isLoading: false,
  items: []
};
List.propTypes = {
  className: _propTypes["default"].string,
  customRowComponent: _propTypes["default"].func,
  isLoading: _propTypes["default"].bool,
  items: _propTypes["default"].instanceOf(Array)
};
var _default = List;
exports["default"] = _default;
