// @buffetjs/core/build/esm/components/Table/ActionCollapse.js
import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';

function ActionCollapse(_ref) {
  var colSpan = _ref.colSpan,
      translatedNumberOfEntry = _ref.translatedNumberOfEntry,
      translatedNumberOfEntries = _ref.translatedNumberOfEntries,
      translatedAction = _ref.translatedAction,
      icon = _ref.icon,
      numberOfSelectedEntries = _ref.numberOfSelectedEntries,
      onConfirm = _ref.onConfirm;
  return /*#__PURE__*/React.createElement("tr", {
    className: "deleteRow"
  }, /*#__PURE__*/React.createElement("td", {
    colSpan: colSpan
  }, /*#__PURE__*/React.createElement("p", null, numberOfSelectedEntries, "\xA0", numberOfSelectedEntries > 1 ? translatedNumberOfEntries : translatedNumberOfEntry, "\xA0 selected\xA0-\xA0", /*#__PURE__*/React.createElement("button", {
    onClick: onConfirm,
    type: "button"
  }, /*#__PURE__*/React.createElement("p", null, translatedAction), icon && /*#__PURE__*/React.createElement(Icon, {
    icon: icon
  })))));
}

ActionCollapse.defaultProps = {
  icon: 'trash',
  translatedNumberOfEntry: 'entry',
  translatedNumberOfEntries: 'entries',
  translatedAction: 'Delete all',
  numberOfSelectedEntries: 0,
  onConfirm: function onConfirm() {}
};
ActionCollapse.propTypes = {
  colSpan: PropTypes.number.isRequired,
  icon: PropTypes.string,
  numberOfSelectedEntries: PropTypes.number,
  onConfirm: PropTypes.func,
  translatedAction: PropTypes.string,
  translatedNumberOfEntries: PropTypes.string,
  translatedNumberOfEntry: PropTypes.string
};
export default ActionCollapse;
