// @buffetjs/core/dist/components/TableHeader/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodash = require("lodash");

var _styles = require("@buffetjs/styles");

var _Icon = _interopRequireDefault(require("../Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function TableHeader(_ref) {
  var headers = _ref.headers,
      onChangeSort = _ref.onChangeSort,
      onSelectAll = _ref.onSelectAll,
      rows = _ref.rows,
      shouldAddCell = _ref.shouldAddCell,
      sortBy = _ref.sortBy,
      sortOrder = _ref.sortOrder,
      withBulkAction = _ref.withBulkAction;
  var checked = rows.length > 0 && rows.every(function (row) {
    return row._isChecked === true;
  });
  var shouldDisplayNotChecked = rows.some(function (row) {
    return row._isChecked === true;
  }) && !checked;
  return _react["default"].createElement("thead", null, _react["default"].createElement("tr", null, withBulkAction && _react["default"].createElement("th", {
    className: "checkCell"
  }, _react["default"].createElement(_styles.Checkbox, {
    onChange: onSelectAll,
    checked: checked,
    someChecked: shouldDisplayNotChecked
  })), headers.map(function (header) {
    var isSortEnabled = header.isSortEnabled,
        name = header.name,
        value = header.value;
    var shouldDisplaySort = isSortEnabled && sortBy === value;
    var firstElementThatCanBeSorted = (0, _lodash.get)(headers.filter(function (h) {
      return h.isSortEnabled;
    }), [0, 'value'], null);
    return _react["default"].createElement("th", {
      key: value,
      onClick: function onClick() {
        onChangeSort({
          sortBy: value,
          firstElementThatCanBeSorted: firstElementThatCanBeSorted,
          isSortEnabled: isSortEnabled
        });
      }
    }, _react["default"].createElement("p", {
      className: isSortEnabled ? 'clickable' : ''
    }, name, shouldDisplaySort && _react["default"].createElement(_Icon["default"], {
      icon: sortOrder || 'asc'
    })));
  }), shouldAddCell && _react["default"].createElement("th", null)));
}

TableHeader.defaultProps = {
  headers: [],
  onChangeSort: function onChangeSort() {},
  onSelectAll: function onSelectAll() {},
  rows: [],
  shouldAddCell: false,
  sortBy: null,
  sortOrder: 'asc',
  withBulkAction: false
};
TableHeader.propTypes = {
  headers: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    isSortEnabled: _propTypes["default"].bool,
    name: _propTypes["default"].string,
    value: _propTypes["default"].string
  })),
  onChangeSort: _propTypes["default"].func,
  onSelectAll: _propTypes["default"].func,
  rows: _propTypes["default"].instanceOf(Array),
  shouldAddCell: _propTypes["default"].bool,
  sortBy: _propTypes["default"].string,
  sortOrder: _propTypes["default"].string,
  withBulkAction: _propTypes["default"].bool
};
var _default = TableHeader;
exports["default"] = _default;
