// @buffetjs/core/dist/components/Enumeration/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _Label = _interopRequireDefault(require("../Label"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Enumeration(_ref) {
  var disabled = _ref.disabled,
      id = _ref.id,
      className = _ref.className,
      name = _ref.name,
      onChange = _ref.onChange,
      options = _ref.options,
      value = _ref.value;
  return _react["default"].createElement(_styles.EnumerationWrapper, {
    className: className
  }, options.map(function (option) {
    return _react["default"].createElement(_Label["default"], {
      key: option.value,
      htmlFor: id || name
    }, _react["default"].createElement(_styles.Enumeration, {
      disabled: disabled,
      checked: option.value === value,
      name: id || name,
      onChange: onChange,
      value: option.value
    }), _react["default"].createElement("span", null, option.label));
  }));
}

Enumeration.defaultProps = {
  className: null,
  disabled: false,
  id: null,
  onChange: function onChange() {},
  options: [],
  value: null
};
Enumeration.propTypes = {
  className: _propTypes["default"].string,
  disabled: _propTypes["default"].bool,
  id: _propTypes["default"].string,
  name: _propTypes["default"].string.isRequired,
  onChange: _propTypes["default"].func,
  options: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    label: _propTypes["default"].string,
    value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].number])
  })),
  value: _propTypes["default"].oneOfType([_propTypes["default"].string, _propTypes["default"].bool, _propTypes["default"].object])
};
var _default = Enumeration;
exports["default"] = _default;
