// @buffetjs/core/dist/components/NavTabs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactRouterDom = require("react-router-dom");

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styles = require("@buffetjs/styles");

var _utils = require("@buffetjs/utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function NavTabs(_ref) {
  var links = _ref.links;
  var color = _styles.colors.greyLink;
  return _react["default"].createElement("ul", null, links.map(function (link, index) {
    color = (0, _utils.gradient)(color, 1.5);
    var content = link.label ? _react["default"].createElement("p", null, link.label) : link.compo || null;
    return _react["default"].createElement(_styles.Tab, {
      key: link.to,
      index: index,
      count: links.length,
      background: color
    }, _react["default"].createElement(_reactRouterDom.NavLink, {
      to: link.to
    }, content));
  }));
}

NavTabs.defaultProps = {
  links: []
};
NavTabs.propTypes = {
  links: _propTypes["default"].arrayOf(_propTypes["default"].shape({
    compo: _propTypes["default"].node,
    label: _propTypes["default"].string,
    to: _propTypes["default"].string
  }))
};
var _default = NavTabs;
exports["default"] = _default;
