// @buffetjs/core/dist/components/Picker/PickerSection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  display: none;\n  position: absolute;\n  top: 38px;\n  left: 0;\n  z-index: 1046;\n  background-color: ", ";\n  border-radius: ", ";\n  border: 1px solid ", ";\n  box-shadow: 0 2px 4px ", ";\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var PickerSection = _styledComponents["default"].div(_templateObject(), function (_ref) {
  var theme = _ref.theme;
  return theme.main.colors.white;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.main.sizes.borderRadius;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.main.colors.darkGrey;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.main.colors.greyAlpha;
}, function (_ref5) {
  var isOpen = _ref5.isOpen;
  return isOpen && 'display: block;';
});

PickerSection.defaultProps = {
  isOpen: false,
  theme: {
    main: {
      colors: {
        white: '#ffffff',
        darkGrey: '#e3e9f3',
        greyAlpha: 'rgba(227, 233, 243, 0.5)'
      },
      sizes: {
        borderRadius: '2px'
      }
    }
  }
};
PickerSection.propTypes = {
  isOpen: _propTypes["default"].bool
};
var _default = PickerSection;
exports["default"] = _default;
