// @buffetjs/core/dist/components/Text/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _colors = _interopRequireDefault(require("../../theme/colors"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n  margin: 0;\n  line-height: ", ";\n  color: ", ";\n  font-size: ", ";\n  font-weight: ", ";\n  text-transform: ", ";\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var Text = _styledComponents["default"].p(_templateObject(), function (_ref) {
  var lineHeight = _ref.lineHeight;
  return lineHeight;
}, function (_ref2) {
  var theme = _ref2.theme,
      color = _ref2.color;
  return theme.main.colors[color] || color;
}, function (_ref3) {
  var theme = _ref3.theme,
      fontSize = _ref3.fontSize;
  return theme.main.sizes.fonts[fontSize];
}, function (_ref4) {
  var theme = _ref4.theme,
      fontWeight = _ref4.fontWeight;
  return theme.main.fontWeights[fontWeight];
}, function (_ref5) {
  var textTransform = _ref5.textTransform;
  return textTransform;
}, function (_ref6) {
  var ellipsis = _ref6.ellipsis;
  return ellipsis && "\n    white-space: nowrap;\n    overflow: hidden;\n    text-overflow: ellipsis;\n  ";
});

Text.defaultProps = {
  color: 'greyDark',
  ellipsis: false,
  fontSize: 'md',
  fontWeight: 'regular',
  lineHeight: 'normal',
  textTransform: 'none',
  theme: {
    main: {
      colors: _colors["default"],
      sizes: {
        fonts: {
          xs: '11px',
          sm: '12px',
          md: '13px',
          lg: '18px',
          xl: '24px'
        }
      },
      fontWeights: {
        regular: 400,
        semiBold: 500,
        bold: 600,
        black: 900
      }
    }
  }
};
Text.propTypes = {
  color: _propTypes["default"].string,
  ellipsis: _propTypes["default"].bool,
  fontSize: _propTypes["default"].string,
  fontWeight: _propTypes["default"].string,
  lineHeight: _propTypes["default"].string,
  textTransform: _propTypes["default"].string,
  theme: _propTypes["default"].object
};
var _default = Text;
exports["default"] = _default;
