// @buffetjs/core/dist/components/Table/ActionCollapse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Icon = _interopRequireDefault(require("../Icon"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ActionCollapse(_ref) {
  var colSpan = _ref.colSpan,
      translatedNumberOfEntry = _ref.translatedNumberOfEntry,
      translatedNumberOfEntries = _ref.translatedNumberOfEntries,
      translatedAction = _ref.translatedAction,
      icon = _ref.icon,
      numberOfSelectedEntries = _ref.numberOfSelectedEntries,
      onConfirm = _ref.onConfirm;
  return _react["default"].createElement("tr", {
    className: "deleteRow"
  }, _react["default"].createElement("td", {
    colSpan: colSpan
  }, _react["default"].createElement("p", null, numberOfSelectedEntries, "\xA0", numberOfSelectedEntries > 1 ? translatedNumberOfEntries : translatedNumberOfEntry, "\xA0 selected\xA0-\xA0", _react["default"].createElement("button", {
    onClick: onConfirm,
    type: "button"
  }, _react["default"].createElement("p", null, translatedAction), icon && _react["default"].createElement(_Icon["default"], {
    icon: icon
  })))));
}

ActionCollapse.defaultProps = {
  icon: 'trash',
  translatedNumberOfEntry: 'entry',
  translatedNumberOfEntries: 'entries',
  translatedAction: 'Delete all',
  numberOfSelectedEntries: 0,
  onConfirm: function onConfirm() {}
};
ActionCollapse.propTypes = {
  colSpan: _propTypes["default"].number.isRequired,
  icon: _propTypes["default"].string,
  numberOfSelectedEntries: _propTypes["default"].number,
  onConfirm: _propTypes["default"].func,
  translatedAction: _propTypes["default"].string,
  translatedNumberOfEntries: _propTypes["default"].string,
  translatedNumberOfEntry: _propTypes["default"].string
};
var _default = ActionCollapse;
exports["default"] = _default;
