// @buffetjs/core/build/esm/components/ListHeader/index.js
/**
 *
 * ListHeader
 *
 */
import React from 'react';
import PropTypes from 'prop-types';
import { ListHeader as StyledListHeader } from '@buffetjs/styles';
import Button from '../Button';
import Flex from '../Flex';
import Padded from '../Padded';
import Text from '../Text';
import BaselineAlignement from './BaselineAlignement';

function ListHeader(_ref) {
  var button = _ref.button,
      title = _ref.title,
      subtitle = _ref.subtitle;
  return /*#__PURE__*/React.createElement(StyledListHeader, null, /*#__PURE__*/React.createElement(Flex, {
    justifyContent: "space-between"
  }, /*#__PURE__*/React.createElement(Padded, {
    top: true,
    bottom: true,
    size: "smd"
  }, /*#__PURE__*/React.createElement(Text, {
    fontSize: "lg",
    fontWeight: "bold",
    lineHeight: "18px"
  }, title), subtitle && /*#__PURE__*/React.createElement(Padded, {
    top: true,
    size: "xs"
  }, /*#__PURE__*/React.createElement(Text, {
    fontSize: "md",
    color: "grey",
    lineHeight: "13px"
  }, subtitle))), button && /*#__PURE__*/React.createElement(BaselineAlignement, null, /*#__PURE__*/React.createElement(Button, button))));
}

ListHeader.defaultProps = {
  button: null,
  title: null,
  subtitle: null
};
ListHeader.propTypes = {
  button: PropTypes.shape({
    color: PropTypes.string,
    icon: PropTypes.bool,
    type: PropTypes.string
  }),
  subtitle: PropTypes.string,
  title: PropTypes.string
};
export default ListHeader;
